addons/content-browse/jaxrs/src/main/java/org/commonjava/indy/content/browse/bind/jaxrs/ContentBrowseResource.java

```diff
@@ -78,43 +80,71 @@
     @Inject
     private UriFormatter uriFormatter;
 
+
     @Inject
     protected JaxRsRequestHelper jaxRsRequestHelper;
 
     @Inject
     private ResponseHelper responseHelper;
 
+    @Inject
+    ContentDigester contentDigester;
+
+    @Context UriInfo uriInfo;
```

Comment:
Do we need to remove @ApplicationScoped from the class for this to work in a multi-threaded way?

Reply:
Right now its not problem multi-threaded that much as returning 
"HTTP/1.1 500 Internal Server Error" on every each http HEAD request to that endpoint ("/api/browse/{packageType}/{type}/{name}/{path: (.*)}") , that's way i'm changing to one method which will handle both routes ( "/" and "/{path}/{path}/... ) with this setup.
For tests yes , i will provide them on next commit iteration for this PR

Reply:
I'm just wondering if we should inject these @Context fields as method parameters to avoid cross-request pollution since the class is application scoped.

Reply:
As far as the tests, if you're still working on this can you mark it as WIP? When the tests come up clean it can be confusing whether it's ready to merge or not.

Reply:
"I'm just wondering if we should inject these @context fields as method parameters to avoid cross-request pollution since the class is application scoped."
- it is one to one relation with every request, problem is ( what i noticed and i search but so far didn't find any specific reason to back up my claim ) when it is put as parameter reference ( im not sure is it @context or is it @httpservletrequest who is doing this ) inside @Head http request it is creating exception ( relating to "i cannot find relating info" this maybe is only specific for our environment ) and thats way i remove it rom @head and place it on object ( which basically should be same ) but on other hand maybe you are right when saying that @applicationscoped is causing thread polution acros requests because we have one instance for each request....  
Yes , i will remove @ApplicationScoped and try without and let you know.

Reply:
"As far as the tests, if you're still working on this can you mark it as WIP? When the tests come up clean it can be confusing whether it's ready to merge or not."
- ok , i will change status right now.